tests/web/rpc/v1/test_endpoint_time_series/test_endpoint_time_series_extrapolation.py

```diff
@@ -0,0 +1,301 @@
+                        avg_sampling_rate=0.0001,
+                    )
+                    for _ in range(len(expected_buckets))
+                ],
+            ),
+        ]
+
+    def test_count_reliable(self) -> None:
+        # store a a test metric with a value of 1, every second for an hour
+        granularity_secs = 120
+        query_duration = 3600
+        store_timeseries(
+            BASE_TIME,
+            1,
+            3600,
+            metrics=[DummyMetric("test_metric", get_value=lambda x: 1)],
+            measurements=[
+                DummyMeasurement(
+                    "client_sample_rate", get_value=lambda s: 1
+                )  # 100% sample rate should result in reliable extrapolation
+            ],
+        )
+
+        message = TimeSeriesRequest(
+            meta=RequestMeta(
+                project_ids=[1, 2, 3],
+                organization_id=1,
+                cogs_category="something",
+                referrer="something",
+                start_timestamp=Timestamp(seconds=int(BASE_TIME.timestamp())),
+                end_timestamp=Timestamp(
+                    seconds=int(BASE_TIME.timestamp() + query_duration)
+                ),
+            ),
+            aggregations=[
+                AttributeAggregation(
+                    aggregate=Function.FUNCTION_COUNT,
+                    key=AttributeKey(type=AttributeKey.TYPE_FLOAT, name="test_metric"),
+                    label="count(test_metric)",
+                    extrapolation_mode=ExtrapolationMode.EXTRAPOLATION_MODE_SAMPLE_WEIGHTED,
+                ),
+            ],
+            granularity_secs=granularity_secs,
+        )
+        response = EndpointTimeSeries().execute(message)
+        expected_buckets = [
+            Timestamp(seconds=int(BASE_TIME.timestamp()) + secs)
+            for secs in range(0, query_duration, granularity_secs)
+        ]
+        assert sorted(response.result_timeseries, key=lambda x: x.label) == [
+            TimeSeries(
+                label="count(test_metric)",
+                buckets=expected_buckets,
+                data_points=[
+                    DataPoint(
+                        data=120,
+                        data_present=True,
+                        reliability=Reliability.RELIABILITY_HIGH,
+                        avg_sampling_rate=1,
+                    )
+                    for _ in range(len(expected_buckets))
+                ],
+            ),
+        ]
+
+
+class TestUtils:
+    @pytest.mark.parametrize(
+        ("row_data", "column_name", "average_sample_rate", "reliability"),
+        [
+            (
+                {
+                    "time": "2024-4-20 16:20:00",
+                    "count(sentry.duration)": 100,
+                    "p95(sentry.duration)": 123456,
+                    CUSTOM_COLUMN_PREFIX
+                    + "upper_confidence$count(sentry.duration)$function_type:count": 20,
```

Comment:
would be good to generate this in the way it would be generated in production, rather than just hardcoding it. what if you change the serialization?

Reply:
it's also clearer to the reader like: "oh this is how this column for this aggregation would be generated, this is then how it would show up in the results